drivers/spi/spi_gecko_eusart.c

```diff
@@ -0,0 +1,363 @@
+/*
+ * Copyright (c) 2024 Daikin Comfort Technologies North America, Inc.
+ * Copyright (c) 2019 Christian Taedcke <hacking@taedcke.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT silabs_gecko_spi_eusart
+
+#define LOG_LEVEL CONFIG_SPI_LOG_LEVEL
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(spi_gecko_eusart);
+#include "spi_context.h"
+
+#include <zephyr/sys/sys_io.h>
+#include <zephyr/device.h>
+#include <zephyr/drivers/spi.h>
+#include <soc.h>
+
+#include "em_cmu.h"
+#include "em_eusart.h"
+
+#include <stdbool.h>
+#include <zephyr/drivers/pinctrl.h>
+
+#ifdef CONFIG_CLOCK_CONTROL
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/clock_control/clock_control_silabs.h>
+#define GET_GECKO_EUSART_CLOCK(idx)                            \
+	.clock_dev = DEVICE_DT_GET(DT_INST_CLOCKS_CTLR(idx)), \
+	.clock_cfg = SILABS_DT_INST_CLOCK_CFG(idx),
+#else
+#if (EUSART_COUNT == 1)
+#define CLOCK_EUSART(ref)	(((ref) == EUSART0) ? cmuClock_EUSART0 \
+			       : -1)
+#elif (EUSART_COUNT == 2)
+#define CLOCK_EUSART(ref)	(((ref) == EUSART0) ? cmuClock_EUSART0 \
+			       : ((ref) == EUSART1) ? cmuClock_EUSART1 \
+			       : -1)
+#elif (EUSART_COUNT == 3)
+#define CLOCK_EUSART(ref)	(((ref) == EUSART0) ? cmuClock_EUSART0 \
+			       : ((ref) == EUSART1) ? cmuClock_EUSART1 \
+			       : ((ref) == EUSART2) ? cmuClock_EUSART2 \
+			       : -1)
+#elif (EUSART_COUNT == 4)
+#define CLOCK_EUSART(ref)	(((ref) == EUSART0) ? cmuClock_EUSART0 \
+			       : ((ref) == EUSART1) ? cmuClock_EUSART1 \
+			       : ((ref) == EUSART2) ? cmuClock_EUSART2 \
+			       : ((ref) == EUSART3) ? cmuClock_EUSART3 \
+			       : -1)
+#else
+#error "Undefined number of USARTs."
+#endif /* USART_COUNT */
+#define GET_GECKO_EUSART_CLOCK(id) \
+	.clock = CLOCK_EUSART((EUSART_TypeDef *)DT_INST_REG_ADDR(id)),
+#endif /*CONFIG_CLOCK_CONTROL*/
```

Comment:
Does it make sense to support case where `!CONFIG_CLOCK_CONTROL`? @gmarull, I assume you have an opinion?

Reply:
@Terezvent I guess having support for `!CONFIG_CLOCK_CONTROL` case allows to ensure compatibility with Silabs Series-1. However, AFAIK, no chip in Series-1 includes EUSART.

Reply:
Ok, I already removed that part of the code, thanks @jerome-pouiller @gmarull